drivers/gpu/drm/msm/adreno/adreno_gpu.c

```diff
@@ -1085,6 +1085,55 @@ int adreno_read_speedbin(struct device *dev, u32 *speedbin)
 	return nvmem_cell_read_variable_le_u32(dev, "speed_bin", speedbin);
 }
 
+static u32 fuse_to_supp_hw(const struct adreno_info *info, u32 fuse)
+{
+	if (!info->speedbins)
+		return UINT_MAX;
+
+	for (int i = 0; info->speedbins[i].fuse != SHRT_MAX; i++)
+		if (info->speedbins[i].fuse == fuse)
+			return BIT(info->speedbins[i].speedbin);
+
+	return UINT_MAX;
+}
+
+int adreno_set_supported_hw(struct device *dev, const struct adreno_info *info)
+{
+	u32 supp_hw;
+	u32 speedbin;
+	int ret;
+
+	ret = adreno_read_speedbin(dev, &speedbin);
+	/*
+	 * -ENOENT means that the platform doesn't support speedbin which is
+	 * fine
+	 */
+	if (ret == -ENOENT) {
+		return 0;
+	} else if (ret) {
+		return dev_err_probe(dev, ret,
+				"failed to read speed-bin. Some OPPs may not be supported by hardware\n");
+	}
+
+	supp_hw = fuse_to_supp_hw(info, speedbin);
+
+	dev_info(dev, "GPU speedbin fuse %d (0x%x), mapped to opp-supp-hw 0x%x\n",
+		 speedbin, speedbin, supp_hw);
```

Comment:
This debug I added myself, this was not originally present in a6xx_gpu before moving, so perhaps it should be removed too

Reply:
I'd better extract this debug print to extra commit that can be easily dropped during some rebase